### ISSUE TITLE
[tests] Consider using some Aspire.Hosting.Testing API in template, and EndToEnd tests

### ISSUE THREAD

**Issue by radical** [dotnet/aspire#4549]
- [ ] For EndToEnd tests we could get the logs from the services, to surface better errors, and have earlier detection of failures (like the container start failing)
- [x] Not applicable as this runs the projects created from templates as-is. ~~For template tests - this needs to be identified~~

**Comment by mitchdenny** [comment]
I think with this stuff if we can get one good example of doing this into the repo folks will start to copy it.

**Comment by davidfowl** [comment]
@radical did we ever do this?

**Comment by radical** [comment]
No, we never got around to this. And `EndToEnd` tests are pretty limited now. Aspire.Hosting.Testing.Tests works better for doing something like this now, IIUC. I'm closing this, but please re-open if we want to explore something here.
